psl-core/src/main/java/org/linqs/psl/util/RandUtils.java

```diff
@@ -176,4 +174,73 @@ public static synchronized void pairedShuffle(List... lists) {
             indexes[swapIndex] = indexes[i];
         }
     }
+
+    /**
+     * Sample from a dirichlet with the provided parameters
+     */
+    public static synchronized double[] sampleDirichlet(double[] alphas) {
+        double [] gammaSamples = new double[alphas.length];
+        double gammaSampleSum = 0;
+        double [] dirichletSample = new double[alphas.length];
+
+        for(int i = 0; i < alphas.length; i++){
+            gammaSamples[i] = nextGamma(alphas[i],1);
+            gammaSampleSum = gammaSampleSum + gammaSamples[i];
+        }
+
+        for(int i = 0; i < alphas.length; i++){
+            dirichletSample[i] = gammaSamples[i] / gammaSampleSum;
+        }
+
+        return dirichletSample;
+    }
+
+    /**
+     * Sample from a gamma distribution with the provided shape and scale parameters.
+     * See Marsaglia and Tsang (2000a): https://dl.acm.org/doi/10.1145/358407.358414
+     */
+    public static synchronized double nextGamma(double shape, double scale) {
+        boolean transform_flag;
+        double alpha;
+        double d;
+        double c;
+        double Z;
+        double V;
+        double U;
+        double gamma_sample;
```

Comment:
I am not a fan of any of these variable names.
For the ones with underscores, use camelCase.
For the uninformative ones (`alpha` and one letter), try to pick a name that is more descriptive and then comment on the variable what it matches up to in the original work.
Ex:
```java
boolean learningRate = false;  // alpha
```
This is a case where it can be okay to not use full sentences.

Also, initialize them all.